src/plots/cartesian/layout_defaults.js

```diff
@@ -173,6 +173,12 @@ module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, fullData) {
 
         handleAxisDefaults(axLayoutIn, axLayoutOut, coerce, defaultOptions, layoutOut);
 
+        coerce('showspikes');
```

Comment:
When `showspikes: false`, the attributes below shouldn't get coerced (as they don't do anything).